app/controllers/application_controller.rb

```diff
@@ -1,4 +1,11 @@
 class ApplicationController < ActionController::Base
   include ApplicationHelper
   before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_in, keys: %i[totp_code cognito_session_id challenge_parameters])
```

Comment:
Should we be letting the user provide the cognito_session_id back to us as form parameters?

Reply:
I have updated this... Originally we were having to embed these variables in the form but have since moved them in to the session.